templates/config.cli/12-syslog.epp

```diff
@@ -0,0 +1,23 @@
+<%- |
+String $prefix,
+Boolean $syslog,
+String $syslog_app_name,
+String $syslog_facility,
+String $syslog_hostname,
+String $syslog_level,
+String $syslog_port,
+Stdlib::Host $syslog_server_address,
+String $syslog_format
+| -%>
+<%- if $syslog { -%>
+if (outcome != success) of <%= $prefix -%>/subsystem=logging/syslog-handler=SYSLOG:read-resource
+<%= $prefix -%>/subsystem=logging/syslog-handler=SYSLOG:add
```

Comment:
The `outcome` condition should end after the `add`. If you move `end-if` to after this line then you ensure that changes to syslog options will actually change the file even if SYSLOG handler is already set.  I am fairly certain that `write-attribute` can be run over and over and not break if already exists.